HarClient.Tests/Mocks/MockServerHandlerBehaviour.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace HarClient.Tests.Mocks
+{
+    /// <summary>
+    /// Meta-tests to assert the behaviour of the mock object.
+    /// </summary>
+    [TestClass]
+    public class MockServerHandlerBehaviour
+    {
+        static MockServerHandler CreateHandler()
```

Comment:
This effectively enumerates all the test cases up-front, which is confusing and effectively "cheats" when we go to write the actual tests.

In the spirit of dogfooding, we should define each inline for each test method.

Reply:
Either that, or consider supplying this as a "baseline" instance with a well-known behaviour.